pkg/scheduler/dependency_graph.go

```diff
@@ -217,28 +236,32 @@ func (sched Scheduler) newScheduledResource(kind, name string, resDefs map[strin
 		return nil, fmt.Errorf("not a proper resource kind: %s. Expected '%s'",
 			kind, strings.Join(resources.Kinds, "', '"))
 	}
-	r, err := sched.newResource(kind, name, resDefs, gc, resourceTemplate)
+	r, existing, err := sched.newResource(kind, name, resDefs, gc, resourceTemplate, silent)
 	if err != nil {
 		return nil, err
 	}
 
-	return newScheduledResourceFor(r, gc), nil
+	return newScheduledResourceFor(r, gc, existing), nil
 }
 
 func (sched Scheduler) newResource(kind, name string, resDefs map[string]client.ResourceDefinition,
-	gc *GraphContext, resourceTemplate interfaces.ResourceTemplate) (interfaces.Resource, error) {
+	gc *GraphContext, resourceTemplate interfaces.ResourceTemplate, silent bool) (interfaces.Resource, bool, error) {
```

Comment:
Sometimes you just need to see how dependency graph would look like in some particular configuration. BuildDependencyGraph is a computational-only method and method which doesn't have any side effects (well, it didn't had them before flows, now it can produce Replica resources if it asked to do so). The same goes for `newResource` which doesn't actually create new resources but instead creates a resource factory (this method existed before the flows, maybe it should have a different name, anyway). Which mean that if `BuildDependencyGraph` can be used just to get the graph without any side effects. However logging is also a side effect and thus there should be a way to suppress it.  In my case I need to get dependency graph in one configuration during construction of another dependency graph. If I not suppress the logging, the user will see a mix of messages from both graphs, many messages will be duplicated because two graphs overlap. Because `newReource` is called from BuildDependencyGraph` I had to add `silent` parameter to it as well